### PR TITLE
First round at removing static httprequest layer, moving to HttpManager

### PR DESCRIPTION
**Pull request by MarkZuber**
Create a new class, HttpManager and its corresponding interface, IHttpManager.  The intent is to replace the static HttpRequest class with this object and bound usage of this to IHttpManager so we can simplify our http layer mocking.  This currently gets us a non-static version of HttpManager and places its implementation under HttpRequest's static calls as the first step.

### CODE REVIEW

File: core/src/Exceptions/CoreExceptionFactory.cs [hunk 1]
```diff
@@ -29,11 +29,42 @@
 
 namespace Microsoft.Identity.Core
 {
+    internal interface ICoreExceptionFactory
```
Comment (henrik-me) [inline core/src/Exceptions/CoreExceptionFactory.cs]: >internal interface ICoreExceptionFactory [](start = 4, length = 40)

separate file?

**Comment by henrik-me** [comment]
aside: as we move away from non-static (good) then we should still ensure the httpc;lient is re-used for the lifetime of our own clientapplication objects

**Review by henrik-me — APPROVED** [review]
:shipit:

**Review by bgavrilMS — APPROVED** [review]
:shipit:

File: core/src/Exceptions/CoreExceptionFactory.cs (hunk 1 above)
Comment (MarkZuber) [reply core/src/Exceptions/CoreExceptionFactory.cs]: fixed

**Pull request merged**